Source/JavaScriptCore/heap/MarkedBlock.cpp

```diff
@@ -577,9 +577,8 @@ NO_RETURN_DUE_TO_CRASH NEVER_INLINE void MarkedBlock::dumpInfoAndCrashForInvalid
         StringPrintStream out;
         out.printf("INVALID HANDLE [%d]: markedBlock=%p; heapCell=%p; cellFirst8Bytes=%#llx; subspaceHash=%#x; contiguousZeros=%lu; totalZeros=%lu; blockVM=%p; actualVM=%p; isBlockVMValid=%d; isBlockInSet=%d; isBlockInDir=%d; foundInBlockVM=%d;",
             line, this, heapCell, cellFirst8Bytes, subspaceHash, contiguousZeroBytesHeadOfBlock, totalZeroBytesInBlock, blockVM, actualVM, isBlockVMValid, isBlockInSet, isBlockInDirectory, foundInBlockVM);
-        const char* msg = out.toCString().data();
-        WTF::setCrashLogMessage(msg);
-        dataLogLn(msg);
+        WTF::setCrashLogMessage(out.toCString().data());
+        dataLogLn(out.toCString().data());
```

Comment:
nit: The issue here is that we produce the CString but don't keep it alive until after we're done with it.  There's no reason to run this CString conversion twice (though it doesn't hurt much).  Why not just do:
```
auto cstring = out.toCString();
WTF::setCrashLogMessage(cstring.data());
dataLogLn(cstring.data());
```